src/Plugin/Field/FieldFormatter/LinkEmbedFormatter.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\url_embed\Plugin\Field\FieldFormatter\LinkEmbedFormatter.
+ */
+
+namespace Drupal\url_embed\Plugin\Field\FieldFormatter;
+
+use Drupal\Component\Utility\SafeMarkup;
+use Drupal\Core\Field\FieldDefinitionInterface;
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Field\FormatterBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\Url;
+use Drupal\link\LinkItemInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Embed\Embed;
+
+/**
+ * Plugin implementation of the 'URL Embed' formatter.
+ *
+ * @FieldFormatter(
+ *   id = "url_embed",
+ *   label = @Translation("Preview URL using Embed library"),
+ *   field_types = {
+ *     "link"
+ *   }
+ * )
+ */
+class LinkEmbedFormatter extends FormatterBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
```

Comment:
I don't think we actually need to implement ContainerFactoryPluginInterface because we don't need anything additional here in the create() method.